src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -111,6 +111,7 @@ public new virtual SqliteConnection Connection
                 {
                     DisposePreparedStatements();
                     _connection = value;
+                    _connection.Commands.Add(new WeakReference<SqliteCommand>(this));
```

Comment:
Need to remove it from the old connection (if any). Otherwise, this might get disposed when it shouldn't be.